src/test/java/com/endava/internship/collections/StudentListTest.java

```diff
@@ -399,26 +417,188 @@ void subListWithInvalidIndexesCaseTwo(int firstIndex) {
     }
 
     @Test
-    void addAll() {
+    void addAllThrowsException() {
+        //given
+        StudentList studentList = null;
+        //when
+        IllegalArgumentException exception =
+                assertThrows(IllegalArgumentException.class,()->students.addAll(studentList),
+                        "Exception should be thrown");
+        //then
+        assertEquals(exception.getMessage(),"Argument, passed to the method is null",
+                "First index should be greater than last");
+    }
+    @Test
+    void addAllWorksNice() {
+        //given
+        Student valik = new Student(names[1],birthdays[2],details[3]);
+        Student vova = new Student(names[1],birthdays[1],details[1]);
+        Student petea = new Student(names[0],birthdays[0],details[0]);
+        StudentList studentList = new StudentList();
+        studentList.add(valik);
+        studentList.add(vova);
+        studentList.add(null);
+        studentList.add(petea);
+        studentList.add(null);
+        //when
+        boolean result= students.addAll(studentList);
+        //then
+        assertAll(
+                ()->assertTrue(result,"Returned true because of operation succeeds"),
+                ()->assertEquals(studentList.get(0),students.get(0),"The order should be like in sourse list"),
+                ()->assertEquals(studentList.get(1),students.get(1),"The order should be like in sourse list"),
+                ()->assertEquals(studentList.get(2),students.get(2),"The order should be like in sourse list")
+        );
     }
 
     @Test
-    void testToString() {
+    void containsAllFails() {
+        //given
+        Student valik = new Student(names[1],birthdays[2],details[3]);
+        Student vova = new Student(names[1],birthdays[1],details[1]);
+        Student petea = new Student(names[0],birthdays[0],details[0]);
+        students.add(valik);
+        students.add(vova);
+        StudentList studentList = new StudentList();
+        studentList.add(valik);
+        studentList.add(vova);
+        studentList.add(petea);
+        //when
```

Comment:
add empty lines before `// when` and `// then`
(everywhere)

Reply:
Fixed!